packages/server/src/__tests__/model-state.test.ts

```diff
@@ -15,6 +15,26 @@ describe('model-state', () => {
     setS((s) => s + 1)
     expect(s.get()).toBe(1)
   })
+  test('mutation setter with no update', () => {
+    const handler = vi.fn()
+    const [s, setS] = state(0)
+    const release = s.subscribe(handler)
+    expect(handler).toBeCalledTimes(1)
+    expect(handler).toHaveBeenLastCalledWith(0)
+    expect(s.get()).toBe(0)
+    setS((s) => s)
+    expect(handler).toBeCalledTimes(1)
+    setS((s) => s + 1)
+    expect(handler).toBeCalledTimes(2)
+    expect(handler).toHaveBeenLastCalledWith(1)
+    release()
+    setS((s) => s + 1)
+    expect(s.get()).toBe(2)
```

Comment:
What happens in between setS invocations that s is always 2? Does "release" function resets it in some way?

Reply:
s starts off as 0 and we increment it twice, that is why s is 2 here

release in this context means "unsubscribe" so we also make sure that the subscription doesn't get called anymore. but that isn't really necessary in this test